### PR TITLE
Count Starting along with Running VMs for user dispersing planner

### PR DESCRIPTION
**Pull request by anuragaw**
Consider running and starting VMs when considering load ona host for VM deployement for more accurate dispersion.

## Description

Fixes: #3442 
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Deploy VMs quickly through CMK on a 2 host environment.

### CODE REVIEW

**Comment by anuragaw** [comment]
ping @rhtyd , @shwstppr , @nvazquez , @borisstoyanov - for review.

File: engine/schema/src/main/java/com/cloud/vm/dao/VMInstanceDaoImpl.java [hunk 1]
```diff
@@ -749,10 +749,10 @@ public Long countActiveByHostId(long hostId) {
     }
 
     @Override
-    public Long countRunningByAccount(long accountId) {
-        SearchCriteria<Long> sc = CountRunningByAccount.create();
+    public Long countRunningAndStartingByAccount(long accountId) {
+        SearchCriteria<Long> sc = CountRunningAndStartingByAccount.create();
         sc.setParameters("account", accountId);
-        sc.setParameters("state", State.Running);
+        sc.setParameters("states", new Object[] {State.Starting, State.Running});
```
Comment (rhtyd) [inline engine/schema/src/main/java/com/cloud/vm/dao/VMInstanceDaoImpl.java]: You can simply also write, `State.Starting, State.Running`

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Failed tests seem unrelated:
```
 test_04_extract_Iso  | Failure              | 128.332 | test_iso.py          |
+----------------------+----------------------+---------+----------------------+
| test_06_copy_iso     | Skipped              | 0       | test_iso.py          |
+----------------------+----------------------+---------+----------------------+
| test_04_rvpc_interna | exceptions.TypeError | 204.492 | test_internal_lb.py  |
| llb_haproxy_stats_on |                      |         |                      |
| _all_interfaces      |                      |         |                      |
+----------------------+----------------------+---------+----------------------+
| test_static_role_acc | Skipped              | 0.023   | test_staticroles.py  |
| ount_acls            |                      |         |                      |
+----------------------+----------------------+---------+----------------------+
| test_change_service_ | Skipped              | 0.001   | test_vm_snapshots.py |
| offering_for_vm_with |                      |         |                      |
| _snapshots           |                      |         |                      |
+----------------------+----------------------+---------+----------------------+
| test_01_redundant_vp | Failure              | 278.981 | test_vpc_vpn.py      |
| c_site2site_vpn      |                      |         |                      
```

**Pull request merged**